docs/proposals/policy.md

```diff
@@ -0,0 +1,258 @@
+ 1.  A Policy is a container object that attempts to mitigate referential integrity problems.  Referential integrity problems **still exist!**  Local RoleBindings can still reference global Roles stored in a different object, so dangling references can still happen.  In usage, a RoleBinding that references a missing Role will produce an error.
+ 1.  Roles versus Teams.  A Role is a grouping of PolicyRules.  A team is a grouping of users.  A RoleBinding associates a Role with a Team.  This distinction makes Teams easier to share across projects.
+
+Policy Evaluation
+In order to determine whether a request is authorized, the AuthorizationAttributes are tested in the following order:
+  1. all allow RoleBinding PolicyRules in the namespace
+  1. all deny RoleBinding PolicyRules in the namespace
+  1. all allow RoleBinding PolicyRules in the global namespace
+  1. all deny RoleBinding PolicyRules in the global namespace
+
+Returns include both the allowing reason and the denying reason.
+
+### Configtime Authorization Types
+These are the types used in the policy example above.  They allow us: to quickly find the policy rules that apply to namespace in etcd, to easily delegate project level control to a project admin, and to easily segregate all role control (a project admin can create/update/delete RoleBindings, but not Roles).
+```
+// PolicyRule holds information that describes a policy rule, but does not contain information 
+// about who the rule applies to or which namespace the rule applies to.
+type PolicyRule struct {
+	// Deny is true if any request matching this rule should be denied.  If false, any request matching this rule is allowed.
+	Deny bool
+	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.  "*" represents all kinds.
+	Verbs []Verb
+	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.  
+	AttributeRestrictions runtime.EmbeddedObject
+	// ResourceKinds is a list of kinds this rule applies to.  "*" represents all kinds.
+	ResourceKinds []string
+}
+
+// Role is a logical grouping of PolicyRules that can be referenced as a unit by RoleBindings.
+type Role struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	Rules []PolicyRule
+}
+
+// RoleBinding references a Role, but not contain it.  It adds who and namespace information.  
+// It can reference any Role in the same namespace or in the global namespace.
+type RoleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+
+	// this is a ref to a Role
+	ReferencesGlobalRole bool
+	RoleName             string
```

Comment:
> Is that explicit in this doc?  How is it enforced as a singleton?

Godoc says it holds all the policy for a particular namespace implying singleton, but I'll update that explicitly.  The API section specifically says that there is no REST endpoint for creating, updating, or deleting Policy resources, so it's enforced by the code backing role and rolebinding modifications.